pkg/inotify-informer/inotify.go

```diff
@@ -0,0 +1,177 @@
+package inotifyinformer
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+
+	"kubevirt.io/kubevirt/pkg/logging"
+	"kubevirt.io/kubevirt/pkg/virt-handler/virtwrap/api"
+)
+
+// NewFileListWatchFromClient creates a ListWatcher which watches for file
+// creations, recreations, and deletions.
+// It is a special ListWatcher, since it can't be used to stay completely
+// in sync with the file system content. Instead it provides at-least-once
```

Comment:
Let's add more concrete limitations: During initial synchronisation or relisting, there is a small possibility for a race condition where file creation and modification events are delivered at least once. Deletes are always exactly once delivered.

Reply:
fixed